src/test/java/edu/ucla/library/prl/harvester/AuthorizedFIT.java

```diff
@@ -0,0 +1,79 @@
+
+package edu.ucla.library.prl.harvester;
+
+import static edu.ucla.library.prl.harvester.handlers.AuthHandler.FORM_PASSWORD;
+import static edu.ucla.library.prl.harvester.handlers.AuthHandler.FORM_USERNAME;
+
+import java.util.List;
+
+import info.freelibrary.util.HTTP;
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientSession;
+import io.vertx.uritemplate.UriTemplate;
+
+/**
+ * A base test class that adds authorization support to the functional tests that extend it.
+ */
+class AuthorizedFIT {
+
+    /** A test logger. */
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizedFIT.class, MessageCodes.BUNDLE);
+
+    /** The login URL template. */
+    private static UriTemplate LOGIN_URL = UriTemplate.of("/login");
+
+    /** The name of the Vert.x session cookie. */
+    private static final String COOKIE_NAME = "vertx-web.session";
+
+    /**
+     * Tries to authorize a new WebClient interaction. The {@link WebClient} passed here must be created with a
+     * {@link WebClientSession}.
+     *
+     * @param aWebClient An unauthorized Web client
+     * @return A successful future with an authorized Web client or a failed future if the authorization failed
+     */
+    protected Future<WebClient> authorize(final WebClient aWebClient) {
```

Comment:
Are you saying you'd prefer this method (and its related method) to be accessed from a utility class instead of an base class that the individual tests extend?

Reply:
I think it should stay here but just be marked static since it doesn't reference any member vars.

Reply:
I'm confused by the utility of marking a method static that's not referenced in a static way(?) Can you point me to something that will give me a more detailed explanation to explain the reasoning?